app/models/internal_post.rb

```diff
@@ -0,0 +1,56 @@
+class InternalPost < Post
+  belongs_to :user
+  validates :user, :title, :body_markdown, presence: true
+  has_many :comments, as: :parent
+
+  before_save do
+    parser = Redcarpet::Markdown.new(Redcarpet::Render::HTML, autolink: true)
+    self.body_html = parser.render(body_markdown)
+  end
+
+  after_create :notify_twitter
+  # TODO: move this to background job
+
+  def notify_twitter
+    if Rails.env.production?
+      $twitter_client.update(tweet_content)
+    end
+  end
+
+  def tweet_content
+    url = Rails.application.routes.url_helpers.
+      post_short_link_url(self, host: "rbga.me")
+    url = " #{url}"
+    max_title_length = 140 - url.length
+    title[0...max_title_length] + url
+  end
+
+  def username
+    self.user.username
+  end
+
+  def number_of_comments
+    comments.count
+  end
+
+  def create_tags_from_tag_string
```

Comment:
External Posts also have tags but we don't need the same method because we put  a tag called Question directly.